src/ngp_algorithms/TurbViscKEAlg.C

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include "ngp_algorithms/TurbViscKEAlg.h"
+#include "ngp_utils/NgpLoopUtils.h"
+#include "ngp_utils/NgpTypes.h"
+#include "ngp_utils/NgpFieldManager.h"
+#include "Realm.h"
+#include "utils/StkHelpers.h"
+#include "stk_mesh/base/MetaData.hpp"
+#include "stk_mesh/base/NgpMesh.hpp"
+
+namespace sierra {
+namespace nalu {
+
+TurbViscKEAlg::TurbViscKEAlg(
+  Realm& realm,
+  stk::mesh::Part* part,
+  ScalarFieldType* tvisc,
+  const bool useAverages)
+  : Algorithm(realm, part),
+    tviscField_(tvisc),
+    density_(get_field_ordinal(realm.meta_data(), "density")),
+    viscosity_(get_field_ordinal(realm.meta_data(), "viscosity")),
+    tke_(get_field_ordinal(realm.meta_data(), "turbulent_ke")),
```

Comment:
we've got some excess fields here we might want to clean up

Reply:
good catch. done